Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM.cpp

```diff
@@ -64,21 +64,24 @@ void FiniteDifferenceSolver::MacroscopicEvolveHMCartesian(
     int M_normalization = warpx.mag_M_normalization;
     int mag_exchange_coupling = warpx.mag_LLG_exchange_coupling;
     int mag_anisotropy_coupling = warpx.mag_LLG_anisotropy_coupling;
-    if (mag_exchange_coupling == 1 || mag_anisotropy_coupling == 1){
-        // H_exchange or H_anisotropy
-        amrex::Abort("No support for the exchange coupling term H_exchange or H_anisotropy in MacroscopicEvolveHM()");
-    }
+    amrex::Vector<amrex::Real> mag_anisotropy_axis = warpx.mag_LLG_anisotropy_axis;
 
     // temporary Multifab storing M from previous timestep (old_time) before updating to M(new_time)
     std::array<std::unique_ptr<amrex::MultiFab>, 3> Mfield_old; // Mfield_old is M(old_time)
 
-    amrex::GpuArray<int, 3> const& mag_Ms_stag    = macroscopic_properties->mag_Ms_IndexType;
-    amrex::GpuArray<int, 3> const& mag_alpha_stag = macroscopic_properties->mag_alpha_IndexType;
-    amrex::GpuArray<int, 3> const& mag_gamma_stag = macroscopic_properties->mag_gamma_IndexType;
-    amrex::GpuArray<int, 3> const& Mx_stag        = macroscopic_properties->Mx_IndexType;
-    amrex::GpuArray<int, 3> const& My_stag        = macroscopic_properties->My_IndexType;
-    amrex::GpuArray<int, 3> const& Mz_stag        = macroscopic_properties->Mz_IndexType;
-    amrex::GpuArray<int, 3> const& macro_cr       = macroscopic_properties->macro_cr_ratio;
+    amrex::GpuArray<int, 3> const& mag_Ms_stag         = macroscopic_properties->mag_Ms_IndexType;
+    amrex::GpuArray<int, 3> const& mag_alpha_stag      = macroscopic_properties->mag_alpha_IndexType;
+    amrex::GpuArray<int, 3> const& mag_gamma_stag      = macroscopic_properties->mag_gamma_IndexType;
+    amrex::GpuArray<int, 3> const& mag_exchange_stag   = macroscopic_properties->mag_exchange_IndexType;
+    amrex::GpuArray<int, 3> const& mag_anisotropy_stag = macroscopic_properties->mag_anisotropy_IndexType;
+    amrex::GpuArray<int, 3> const& Mx_stag             = macroscopic_properties->Mx_IndexType;
+    amrex::GpuArray<int, 3> const& My_stag             = macroscopic_properties->My_IndexType;
+    amrex::GpuArray<int, 3> const& Mz_stag             = macroscopic_properties->Mz_IndexType;
+    amrex::GpuArray<int, 3> const& macro_cr            = macroscopic_properties->macro_cr_ratio;
+    amrex::GpuArray<amrex::Real, 3> anisotropy_axis{0.0,0.0,0.0};
+    for (int i = 0; i < 3; i++){
+        anisotropy_axis[i] = mag_anisotropy_axis[i];
+    }
```

Comment:
SImilar to the other properties, this parameter must be defined in MacroscopicProperties.H
and initialized in MacroscopicProperties.cpp similar to the other properties used above.